VMs/metrics/ubuntu16.04/user-data.proxy

```diff
@@ -0,0 +1,108 @@
+    shell: /bin/bash
+    sudo: ALL=(ALL) NOPASSWD:ALL
+    ssh-authorized-keys:
+      - PUBLICKEYCONTENTS
+
+packages:
+ - gawk
+ - bc
+ - curl
+ - git
+ - jq
+ - default-jre
+ - gcc
+ - make
+ - build-essential
+## PROXY specific mods
+# We need socat for our git proxy script
+ - socat
+
+runcmd:
+ - rm /etc/update-motd.d/10-help-text /etc/update-motd.d/51-cloudguest
+ - rm /etc/update-motd.d/90-updates-available
+ - rm /etc/legal
+## PROXY specific mods
+# Looks like the write_file to the jenkins user homedir, even though the file
+# is owned by jenkins, makes all the dirs owned by root - so, let's put them
+# back
+ - chown -R jenkins /home/jenkins
+ - chgrp -R jenkins /home/jenkins
+
+# Install Go
+# The sourceing of environment is a workaround to get any proxy settings we need
+ - 'bash -c "set -a; source /etc/environment; set +a; curl -L https://dl.google.com/go/go1.10.2.linux-amd64.tar.gz -o /tmp/go1.10.2.linux-amd64.tar.gz"'
+ - tar -C /usr/local -xzf /tmp/go1.10.2.linux-amd64.tar.gz
+ - rm /tmp/go1.10.2.linux-amd64.tar.gz
+
+# Install docker
+# The sourceing of environment is a workaround to get any proxy settings we need
+ - 'bash -c "set -a; source /etc/environment; set +a; curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -"'
+ - sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+ - apt-get update
+ - apt-get install -y docker-ce
+ - gpasswd -a USERNAME docker
+
+# Cleanup
+ - apt-get auto-remove -y
+
+# We leave the VM running, for the user to detach and then run the
+# host side completion script, which will then shut the VM down when
+# finally configured
+
+## PROXY specific mods
+apt:
+    http_proxy: http://PROXY_IP:PROXY_PORT
+    https_proxy: http://PROXY_IP:PROXY_PORT
+    ftp_proxy: http://PROXY_IP:PROXY_PORT
+
+write_files:
+-   content: |
+        http_proxy="http://PROXY_IP:PROXY_PORT"
+        https_proxy="http://PROXY_IP:PROXY_PORT"
+        ftp_proxy="http://PROXY_IP:PROXY_PORT"
+        no_proxy="localhost,127.0.0.1"
+    path: /etc/environment
+    permissions: '0644'
+-   content: |
+        #!/bin/bash
+        PROXY=PROXY_IP
+        exec socat STDIO SOCKS4:$PROXY:$1:$2
+    path: /home/jenkins/bin/gitproxy
+# Just can't get these ownerships to work with cloud-init
+# we'll fix them in the runcmd section.
+#    owner: jenkins:jenkins
+    permissions: '0744'
+-   content: |
+        [core]
+            gitproxy = /home/jenkins/bin/gitproxy
+        [http]
+            proxy = http://PROXY_IP:PROXY_PORT
+        [https]
+            proxy = http://PROXY_IP:PROXY_PORT
+    path: /home/jenkins/.gitconfig
+# Just can't get these ownerships to work with cloud-init
+# we'll fix them in the runcmd section.
+#    owner: jenkins:jenkins
+    permissions: '0644'
+-   content: |
+        [Service]
+        Environment="HTTP_PROXY=http://PROXY_IP:PROXY_PORT"
+        Environment="HTTPS_PROXY=http://PROXY_IP:PROXY_PORT"
```

Comment:
Shouldn't the value be `https` here?

Reply:
That is site dependent. On the site I tested this at, no, it is not. I'll tweak it to be a bit more regexp though

Reply:
Is it normal to have an https proxy variable pointing at an http proxy then?